taskoutput/task_log.go

```diff
@@ -80,29 +71,27 @@ func (o TaskLogs) Get(ctx context.Context, env evergreen.Environment, taskOpts T
 	})
 }
 
-func (o TaskLogs) getLogName(taskOpts TaskOptions, logType TaskLogType) string {
+func (o TaskLogOutput) getLogName(taskOpts TaskOptions, logType TaskLogType) string {
 	prefix := fmt.Sprintf("%s/%s/%d/%s", taskOpts.ProjectID, taskOpts.TaskID, taskOpts.Execution, o.ID())
 
-	var filename string
+	var logTypePrefix string
 	switch logType {
 	case TaskLogTypeAgent:
-		filename = "agent"
+		logTypePrefix = "agent"
 	case TaskLogTypeSystem:
-		filename = "system"
+		logTypePrefix = "system"
 	case TaskLogTypeTask:
-		filename = "task"
+		logTypePrefix = "task"
```

Comment:
Is there a reason we would prefer using special almost-the-same-but-not-quite strings instead of just using the `TaskLogType` string value in the prefix? Seems like it could cut down on some of the duplication.

Reply:
I did this to make the log prefix generation explicit. This may just be me being paranoid, but my fear is that using the `TaskLogType` string values can leave us vulnerable to bugs. For example, someone could easily come along in the future and change the string values without realizing they are used to create the prefix for log storage. This would break log fetching and any other potential downstream systems that use S3 directly (like, potentially, a data warehouse). And for posterity, it may not be immediately obvious why that is happening which can lead to a time consuming investigation and fix.

Reply:
Makes sense, and sounds good!